snapcraft/plugins/go.py

```diff
@@ -101,7 +102,13 @@ def get_pull_properties(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
-        self.build_packages.append('golang-go')
+
+        # Do not try to install golang if it is already available in our PATH.
+        try:
+            subprocess.check_output(['go', 'version'])
+        except OSError:
+            self.build_packages.append('golang-go')
```

Comment:
@chrisglass today, using the remote go part, like in here:
https://github.com/elopio/hashicorp-snaps/blob/master/vault/snap/snapcraft.yaml#L30

That it's a little slower because it compiles go. So in the near future, you would do it with the go snap published by mwhudson. Something like `build-snaps: [go]`.